public/course_lecture.php

```diff
@@ -0,0 +1,211 @@
+    echo "<br>" . $error->getMessage();
+}
+
+
+try {
+    $course_section_students = CourseSectionStudent::includes(["user", "course_section"])->getAll();
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $marked_entities = MarkedEntity::where(array("course_offering_id" => $course_page_id));
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $course_sections = CourseSection::where(array('offering_id' => $course_page_id));
+
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $teams = Team::where(array('course_offering_id' => $course_page_id));
+
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+?>
+
+<?php include "templates/header.php"; ?>
+
+<?php if(get_current_role() == 'instructor') { ?>
+    <h2>Student List </h2>
+    <?php
+    $course_students = array();
+
+    foreach ($course_sections as $section) {
+        try {
+            $course_section_student = CourseSectionStudent::includes("user")->where(array('section_id' => $section->id));
+            foreach ($course_section_student as $student){
+                array_push($course_students, $student->user);
+            }
+            
+        } catch (PDOException $error) {
+            echo "<br>" . $error->getMessage();
+        }
+    }
+
+    if ($course_students && count($course_students)) { ?>
+    <table>
+            <thead>
+                <tr>
+                    <th>#</th>
+                    <th>First Name</th>
+                    <th>Last Name</th>
+                    <th>Email Address</th>
+                    <th>Created At</th>
+                </tr>
+            </thead>
+            <tbody>
+        <?php foreach ($course_students as $row) { ?>
+            <tr>
+                <td><?php echo escape($row->id); ?></td>
+                <td><?php echo escape($row->first_name); ?></td>
+                <td><?php echo escape($row->last_name); ?></td>
+                <td><?php echo escape($row->email); ?></td>
+                <td><?php echo escape($row->created_at);  ?> </td>
+            </tr>
+        <?php } ?>
+        </tbody>
+    </table>
+    <?php } else { ?>
+        <blockquote>No Students found for this course.</blockquote>
+    <?php }?>
+
+    <h2>Teams</h2>
+    <?php if ($teams && count($teams)) { ?>
+    
+        <?php foreach ($teams as $row) { ?>
+           <h3><?php echo "Team # ".$row->id ?></h3>
+           <table>
+                <thead>
+                    <tr>
+                        <th>#</th>
+                        <th>First Name</th>
+                        <th>Last Name</th>
+                        <th>Email Address</th>
+                        <th>Created At</th>
+                    </tr>
+                </thead>
+                <tbody>
+                <?php $students_in_team = TeamMember::includes("user")->where(array('team_id' => $row->id)); 
```

Comment:
why don't you preload all this stuff along with the teams?

Reply:
done